monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,324 @@
+from shapely.geometry import Point, Polygon
+import shapely.geometry
+from pyproj import Geod, Transformer
+import json
+from pathlib import Path
+from typing import List, NamedTuple
+import arrow
+import datetime
+from datetime import datetime, timedelta
+import uuid
+
+class QueryBoundingBox(NamedTuple):
+    ''' This is the object that stores details of query bounding box '''
+    name: str
+    shape: Polygon
+    time_step_after: int    
+    time_step_before: int
+    
+class FlightPoint(NamedTuple):
+    ''' This is the object that stores details of query bounding box '''
+    lat: float
+    lng: float
+    alt: float
+
+class AircraftPosition(NamedTuple):
+    ''' A object to hold AircraftPosition for Remote ID purposes. For more information see, the definition in the standard: https://github.com/uastech/standards/blob/36e7ea23a010ff91053f82ac4f6a9bfc698503f9/remoteid/canonical.yaml#L1091'''
+
+    lat : float
+    lng : float
+    alt : float
+    accuracy_h : str
+    accuracy_v : str
+    extrapolated: int
+    pressure_altitude : int
+
+class AdjacentCircularFlightPathsGenerator():
+
+    ''' A class to generate Flight Paths given a bounding box, this is the main module to generate flight path datasets, the data is generated as latitude / longitude pairs with assoiated with the flights. Additional flight metadata e.g. flight id, altitude, registration number can also be generated '''
+
+    def __init__(self, minx: float, miny: float, maxx: float, maxy: float) -> None: 
+
+        """ Create an AdjacentCircularFlightPathsGenerator with the specified bounding box.
+
+            Once these extents are specified, a grid will be created with two rows.  The idea is that multiple flights tracks will be created within the extents.
+            Args:
+            minx: Western edge of bounding box (degrees longitude)
+            maxx: Eastern edge of bounding box (degrees longitude)
+            miny: Southern edge of bounding box (degrees latitude)
+            maxy: Northern edge of bounding box (degrees latitude)
+
+            Raises:
+            ValueError: If bounding box has more area than a 500m x 500m square.
+        """
+
+        self.minx = minx 
+        self.miny = miny
+        self.maxx = maxx
+        self.maxy = maxy                
+        
+        self.flight_points: List[FlightPoint] = []   # This is a object that containts multiple lists of flight tracks as points, in the latitude, longitude, altitude in tuple format. Depending on how the grid is generated in this case 3 columns and 2 rows with six flight tracks there will be six lists in this object
```

Comment:
I have now updated the code with this change, it meant changing references at a bunch of places including at the RIDAircraftStateWriter class and creating a new `generate_flight_grid_and_path_points` so that the GridCellFlight object is generated and populated in one go.